stdlib/public/runtime/MutexWin32.cpp

```diff
@@ -0,0 +1,90 @@
+//===--- Mutex.cpp - ------------------------------------------------------===//
```

Comment:
It may make sense to hoist all of this implementation into the MutexWin32.h since it is so simple (leaving behind `ConditionPlatformHelper::wait` assuming you add fatalReporting to it).